### PR TITLE
Database

### PR DESCRIPTION
**Pull request by jase-d**
Went with plan B and used a CSV file. The efficiency difference is pretty crazy, my original strategy couldn't get past 2M and was rough towards the end taking 15+ mins then crashing. Although a little tricky at first, highly recommend it if your computer doesn't have a high memory capacity like mine.

Currently only stress tested Postgres.

@hongmm @mbbph @ClaySamuelC

### CODE REVIEW

**Review by hongmm — APPROVED** [review]
@jase-d LGTM. Question, why do you need both psql1 and psql2 files? Also, it looks like you have run cassandra in your schema file for yous psql schema, does cassandra use the same sql language? Remember to add newlines at the bottom of each file too!

**Comment by jase-d** [comment]
> @jase-d LGTM. Question, why do you need both psql1 and psql2 files? Also, it looks like you have run cassandra in your schema file for yous psql schema, does cassandra use the same sql language? Remember to add newlines at the bottom of each file too!

So in total I need 50M reviews. So my plan is to have two separate databases that will carry half the reviews in each. The second file will be connected to a separate ip/port once on an EC2 Micro Instance. Not sure if it's necessary, but I think it might help plus think it's just good practice for me to think of scaling horizontally now, even if it may not be the best solution (Will need to study the subject more).

Also, Cassandra uses Cassandra query language or CQL, and it's extremely similar to a SELECT query language or SQL with regards on how to insert. The only difference I've seen is that when retrieving you cannot use the keyword LIKE to compare, and possibly can't use join tables, which I am not planning on using.

Bottom spaces, will definitely do. Can you remind me of the importance of that? I think it's in the ABB style guide, and want to be consistent either way, but still curious if it has an advantage behind the scenes.

**Comment by hongmm** [comment]
Your reasoning definitely makes sense. I was not aware that CQL was that similar to SQL,  I was not able to get enough research done yesterday on Cassandra so makes perfect sense to keep a similar schema. 

As far as I know, adding blank space is more to keep your code looking clean and a widespread standard across developers. It's similar to how in like Google Docs or Word docs there is some empty whitespace reserved at the top/bottom of the file, just makes it easier to read without feeling like everything is squished together. It is a common enough that github specifically notes this so it is probably a good idea to get in the habit of doing this (eslint catches this too but github specifically notes if you have a EOF line or not)

**Comment by jase-d** [comment]
Thanks a lot for that explanation, I’ll have to make it a habit then!

**Review by jase-d — COMMENTED** [review]
More work on the database branch going to look into a way to run the queries as a simple script from terminal for Cassandra instead of copying and pasting.

There may have been a loss of 5000 / 25M records with Cassandra's COPY method, I think it was the last 5000 going by the logs. I think there are ways to ensure no losses but I've heard that is a downfall of Cassandra.
@hongmm @mbbph @ClaySamuelC

**Review by hongmm — APPROVED** [review]
@jase-d I dont know much about the Cassandra stuff yet and how your setImmediate works but assuming that it behaves as you expect it to, I think the code LGTM! I can double check tomorrow or sometime later once I get into it but for now with the knowledge I have, code looks fine.

**Comment by jase-d** [comment]
Hey all, with my last commit I added a timer function that I think could be cool to collaborate on with everyone. Right now it just has the method I mentioned earlier for 'seconds' in the meeting. But if it is helpful, I was think we could expand on it a bit and create a function that we could re-use in the long term by just cloning it into any project as a library. Feel free to check it out and use it, and collaborate!
@hongmm @mbbph @ClaySamuelC

File: server/postgresql/psqlBulkLoad.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const {Timer} = require('../../point_in_time_v1/timer.js');
+const StopWatch = new Timer()
+const { Client } = require('pg');
+const client = new Client({
+  user: 'username',
+  host: 'localhost',
+  database: 'sdc',
+  password: 'password123',
```
Comment (mbbph) [inline server/postgresql/psqlBulkLoad.js:8]: ditto as above

File: point_in_time_v1/timer.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return this.timeCapsule;
+  };
+
+  end = (timeCapsule) => {
+    let start = timeCapsule.moment;
+    let end = this.moment('seconds');
+    return `FINISHED IN ${Math.floor((end - start) / 60)} MIN ${Math.floor((end  - start) % 60)} SECONDS`
+
+  };
+}
```
Comment (mbbph) [inline point_in_time_v1/timer.js:27]: This is super neat! Packaging it up into a library like you said would be great practice

File: server/postgresql/psql1.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+
+const { Client } = require('pg');
+const client = new Client({
+  user: 'username',
+  host: 'localhost',
+  database: 'sdc',
+  password: 'password123',
```
Comment (mbbph) [inline server/postgresql/psql1.js:7]: may be best practice to place pwd into gitignore

**Review by mbbph — APPROVED** [review]

File: server/postgresql/psql1.js (hunk 3 above)
Comment (jase-d) [reply server/postgresql/psql1.js:7]: Thanks for catching that, I'll add that into my next commit.

**Pull request merged**